File: app/src/main/java/org/uadaf/app/internal/view/layouts/SquareCardViewLayout.java
package org.uadaf.app.internal.view.layouts;

import android.content.Context;
import android.util.AttributeSet;
import androidx.cardview.widget.CardView;

public class SquareCardViewLayout extends CardView {
    public SquareCardViewLayout(Context context) {
        super(context);
    }

    public SquareCardViewLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SquareCardViewLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, widthMeasureSpec);
    }
}

File: app/src/main/java/org/uadaf/app/internal/view/layouts/SquareCardViewLayout.kt
package org.uadaf.app.internal.view.layouts

import android.content.Context
import android.util.AttributeSet
import androidx.cardview.widget.CardView

open class SquareCardViewLayout: CardView {

    constructor(context: Context): super(context)

    constructor(context: Context, attrs: AttributeSet): super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyles: Int): super(context, attrs, defStyles)

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, widthMeasureSpec)
    }

}